.changeset/calm-items-listen.md

```diff
@@ -0,0 +1,14 @@
+---
+"react-router": patch
+"@remix-run/router": patch
+---
+
+`useLocation` hook now accepts a state as generic.
+
+So, you are able to define
```

Comment:
Thus, if you've defined a state you want to send with navigate ```
navigate(
  'thepath', { state: { from: 'yourMessage' } }) // string | boolean | number ...
```
you can retrieve this state in your target component like this: